### PR TITLE
Make `parse_lines` a private interface

### PR DESCRIPTION
**Pull request by xiangce**
- rename `parse_lines` to `_parse_line`, and let it parse  only one line each time
- make it as a private interface of the class, to avoid parsing `lines`
  which are not belongs to the instance of the class object
- add `_parse_line` for the base class
- let `get` and `get_after` return parsed line directly
  then,  no need to call `parse_lines` explicitly, e.g.
  CLS.parse_lines(obj.get_after(xxx))
- uniform the `raw_message` as the key for the raw lines

Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### CODE REVIEW

**Comment by xiangce** [comment]
Hey @bfahr, this PR breaks the plugin repo which I submitted another PR https://github.com/RedHatInsights/insights-plugins/pull/436 to fix it. 
If you think it's OK, please merge them together.

OR. If you think it's not fit to the 1.x branch, I will create another PR only against the master branch.

**Comment by xiangce** [comment]
I'd like to add another `get_parsed_lines` to return the parsed lines instead of refactoring the original `get` method. Working..., but any suggestions are welcome :)

**Comment by xiangce** [comment]
It's ready to review :smile:

File: insights/parsers/krb5kdc_log.py [hunk 1]
```diff
@@ -52,29 +52,20 @@ def get(self, keyword):
...
-        Yields:
-            (dict): A dictionary for each parsed line in the input.
         '''
+
         line_re = re.compile(
             r'(?P<timestamp>\w{3} \d\d \d\d:\d\d:\d\d) ' +
             r'(?P<system>\w\S+) ' +
             r'(?P<service>\w+)\[(?P<pid>\d+)\]\((?P<level>\w+)\): ' +
             r'(?P<message>.*)'
         )
```
Comment (JoySnow) [inline insights/parsers/krb5kdc_log.py]: Are there any necessaries for putting this `line_re` as a class constant value ?
We have examples at class ServerXMLRPCLog in parser rhn_logs.py.

File: insights/parsers/openvswitch_logs.py [hunk 2]
```diff
@@ -42,22 +44,16 @@ class OpenVSwitchLog(LogFileOutput):
...
     time_format = '%Y-%m-%dT%H:%M:%S'
 
-    def get(self, search):
+    def _parse_line(self, line):
         """
-        Find lines with the search string in them, and return a list of
-        dictionaries with the line split up into fields.
+        Parse line into fields.
         """
         fieldnames = ['timestamp', 'sequence', 'module', 'level', 'message']
```
Comment (JoySnow) [inline insights/parsers/openvswitch_logs.py]: Also this `fieldnames` ?

File: insights/parsers/sssd_logs.py [hunk 3]
```diff
@@ -44,58 +44,63 @@ class SSSDLog(LogFileOutput):
     Examples:
 
         >>> logs = shared[SSSDLog]
-
         >>> hups = logs.get("SIGHUP")
```
Comment (JoySnow) [inline insights/parsers/sssd_logs.py]: Maybe this line-47 should change to    
`hups = logs.get_parsed_line("SIGHUP")`

File: insights/core/__init__.py [hunk 4]
```diff
@@ -694,17 +737,23 @@ def test_all_parsers(logstamp):
                 if logstamp >= timestamp:
                     # Later - include
                     including_lines = True
-                    yield line
+                    if parsed_line:
+                        yield parsed_line
+                    else:
+                        yield self._parse_line(line)
```
Comment (bfahr) [inline insights/core/__init__.py]: @xiangce I don't think you can make this change as it changes the default api for the 1.x branch.  It needs to return the same time as the input type to maintain compatibility.

File: insights/core/__init__.py [hunk 5]
```diff
@@ -694,17 +737,23 @@ def test_all_parsers(logstamp):
...
                     # Earlier - start excluding
                     including_lines = False
             else:
                 # If we're including lines, add this continuation line
                 if including_lines:
-                    yield line
+                    if parsed_line:
+                        yield parsed_line
+                    else:
+                        yield self._parse_line(line)
```
Comment (bfahr) [inline insights/core/__init__.py]: Same issue as above.

File: insights/core/__init__.py (hunk 4 above)
Comment (xiangce) [reply insights/core/__init__.py]: Sure, will revert them in the 1.x branch.

File: insights/parsers/sssd_logs.py [hunk 6]
```diff
@@ -44,58 +44,63 @@ class SSSDLog(LogFileOutput):
...
+            'message': 'Received SIGHUP.'
+            'raw_message': '(Tue Feb 14 09:45:02 2017) [sssd] [monitor_hup] (0x0020): Received SIGHUP.'
+        }
 
     """
     time_format = '%b %d %H:%M:%S %Y'
 
-    @staticmethod
-    def parse_lines(lines):
+    def _parse_line(self, line):
```
Comment (bfahr) [inline insights/parsers/sssd_logs.py]: Same comments as above regarding backward compatibility.

File: insights/parsers/vdsm_log.py [hunk 7]
```diff
@@ -42,18 +41,16 @@ class VDSMLog(LogFileOutput):
         }
     """
 
-    def parse_lines(self, lines):
+    def _parse_line(self, line):
```
Comment (bfahr) [inline insights/parsers/vdsm_log.py]: Same comments as above regarding backward compatibility.

File: insights/__init__.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 from .core import YAMLParser  # noqa: F401
 from .core import AttributeDict  # noqa: F401
 from .core import fava  # noqa: F401
-from .core import Syslog  # noqa: F401
+from .core import SysLog  # noqa: F401
```
Comment (bfahr) [inline insights/__init__.py]: See comment below about renaming.

File: insights/core/__init__.py [hunk 9]
```diff
@@ -456,9 +456,49 @@ def __contains__(self, s):
...
+            `keywords`
+        """
+        ret = []
+        for l in self.lines:
+            if ((type(keywords) == list and all(w in l for w in keywords)) or
+                    (type(keywords) == str and keywords in l)):
+                ret.append(self._parse_line(l))
+        return ret
+
+    def get(self, keywords):
```
Comment (bfahr) [inline insights/core/__init__.py]: See my comment below about changing parameter names.

**Comment by xiangce** [comment]
Thanks, @bfahr, I will close this PR since some of the modifications are trying to modify API. Will make another one and retain all the original APIs.

**Pull request closed without merging**